cmis_web_proxy_alf/static/src/cmis_object_wrapper_service.js

```diff
@@ -0,0 +1,38 @@
+/** @odoo-module **/
+
+/* ---------------------------------------------------------
++ * Odoo cmis_web
++ * Authors Laurent Mignon 2016, Maxime Franco 2023 Acsone SA/NV
++ * License in __openerp__.py at root level of the module
++ *---------------------------------------------------------
++*/
+
+import {CmisObjectWrapper} from "@cmis_web/cmis_object_wrapper_service";
+import {patch} from "@web/core/utils/patch";
+
+patch(CmisObjectWrapper.prototype, "alfresco_proxy_alf_url", {
+    setup(cmisObject, cmisSession, params) {
+        this._super(...arguments);
+        this.alfrescoApiLocation = params.alfrescoApiLocation;
+    },
+    getPreviewUrl() {
+        var _url = this._super(...arguments);
+        if (_url) {
+            return _url;
+        }
+        var params = {
+            c: "force",
+            lastModified: "pdf%" + new Date().getUTCMilliseconds(),
+            token: this.token,
```

Comment:
Is `this.token` set anywhere ?

Reply:
As i explain in https://github.com/acsone/alfodoo/pull/182#discussion_r1439306763 the token is generated during the initialization of the CmisFolder field thanks to the setRootFolderId function.